### ISSUE TITLE
INameResolver for Queue Trigger in Azure Function V4

### ISSUE THREAD

**Issue by ranouf** [Azure/Azure-Functions#2143]
Hi,

I would like to use the INameResolver for queue trigger like it's available in Azure WebJob. I know it was not available in the past, but is it plan to add for Azure Function V4? Is there a workaround to be able to use it?

here is the code used for Azure WebJob:

   ```
 public class TriggerNameResolver : INameResolver
    {
        private readonly IConfiguration _configuration;
        private readonly MessagerSettings _messagerSettings;

        public TriggerNameResolver(
            IConfiguration configuration,
            IOptions<MessagerSettings> messagerSettings
        )
        {
            _configuration = configuration;
            _messagerSettings = messagerSettings.Value;
        }

        public string Resolve(string name)
        {
            if (name.Contains(':'))
            {
                return _configuration[name].ToString();
            }
            return (_messagerSettings.QueueName + name).ToLower();
        }
    }
```

Thanks for your help

**Comment by v-bbalaiagar** [comment]
Hi @pragnagopa , Can we consider this issue as a feature request

**Comment by pragnagopa** [comment]
Tagging @fabiocav for triage.

**Comment by theo-albers** [comment]
Yes please, because it doesn't seem to be available. I would like to use IConfiguration with Azure App Service to inject configuration in a more flexible manner.

**Comment by bretthacker** [comment]
Is there an update on this? I'm trying it with a dotnet-isolated function even but the functions (and their queue attributes) seem to fire before my INameResolver fires.

**Comment by JohnWilson-PandSD** [comment]
There is another conversation going on about this over here: https://github.com/Azure/azure-functions-dotnet-worker/issues/393 - I've run into the same issue, and given that .NET 8 is forcing a move to the isolated process model, it would be nice to get it resolved ;)

**Comment by davcarrier** [comment]
Is there any update on this? It's a blocker for us to move to the isolated model

**Comment by theo-albers** [comment]
What works for us: setup IConfiguration, use a connection string name that maps to an IConfiguration value. We pass a tenant ID via the message. So we can't switch service bus connection dynamically, but we can:
- read the service bus connection from config
- setup a tenant context via middleware to execute the trigger for a tenant.

It looks like the isolated model has the service bus listener in the host and the trigger is being executed in the isolated process. That's why the old namereolver solution no longer works.

I really don't understand why service bus connection switching wasn't supported from the start since multi tenancy, tenant isolation is what cloud is all about.

**Comment by qualitas-software** [comment]
I only needed the default resolution part in my startup code, not the custom resolver put into DI and getting the host to use that.

So, I ended up writing my own simple resolver as part of my pre-existing extension method:

    public static string GetResolvedValue(this IConfiguration configuration, string configurationKey)
    {
        var value = configuration[configurationKey];

        if (value?.Length > 2 && value[0] == '%' && value[^1] == '%') // resolve?
            value = configuration[value[1..^1]];

        return value;
    }

It will only resolve the whole string, ie- a value that starts and ends with `%`

**Comment by borigas** [comment]
@qualitas-software how are you configuring it to use your `GetResolvedValue()`?
